packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -337,6 +338,37 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
+    // Add the dependencies on other plugin projects to the plugin projects.
+    // A plugin A can depend on plugin B. As a result, this dependency must be surfaced by
+    // making the Gradle plugin project A depend on the Gradle plugin project B.
+    private void configurePluginDependencies(String pluginName, String dependencies) {
+        Project pluginProject = project.rootProject.findProject(":$pluginName")
+        if (pluginProject == null) {
+            // Ignore plugins that don't have a project since most likely they don't
+            // have an android/ directory.
+            return
+        }
+        dependencies.split(',').each { pluginDependencyName ->
+            if (pluginDependencyName.empty) {
+                return
+            }
+            Project dependencyProject = project.rootProject.findProject(":$pluginDependencyName")
+            if (dependencyProject == null) {
+                return
+            }
+            // Wait for the Android plugin to load and add the dependency to the plugin project.
+            pluginProject.afterEvaluate {
+                pluginProject.dependencies {
+                    if (project.getConfigurations().findByName("implementation")) {
```

Comment:
Add a code comment for what this is for. Is it just to handle the old AGP APIs the users use and match them?

Reply:
this isn't needed actually.